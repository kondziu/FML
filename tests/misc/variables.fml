// Expected output:
// > x=1, y=2
// > x=0, y=2
// > x=3, y=2, z=3
// > x=3, y=5, z=4
// > x=3, y=5, z=3

let x = 1;
let y = 2;

print("x=~, y=~\n", x, y);

x <- 0;

print("x=~, y=~\n", x, y);

x <- (let z = 3);

print("x=~, y=~, z=~\n", x, y, z);

begin
    let z = 4;
    y <- 5;
    print("x=~, y=~, z=~\n", x, y, z);
end;

print("x=~, y=~, z=~\n", x, y, z);
