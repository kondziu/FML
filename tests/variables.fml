// Expected output:
// > null
// > false true
// > -42 -1 0 1 42
print("~\n", null);
print("~ ~\n", false, true);
print("~ ~ ~ ~ ~\n", -42, -1, 0, 1, 42);