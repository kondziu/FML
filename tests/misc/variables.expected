x=1, y=2
x=0, y=2
x=3, y=2, z=3
x=3, y=5, z=4
x=3, y=5, z=3
